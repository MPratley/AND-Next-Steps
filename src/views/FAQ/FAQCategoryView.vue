<template>
  <div class="category">
    <h2 class="title">Categories</h2>
    <div class="row">
      <div class="column">
        <div class="category-column" @click="() => selectCategory('Club')">
          <i class="material-icons category-icon" style="padding:35px">home</i>
          <br />Club
        </div>
      </div>
      <div class="column">
        <div class="category-column" @click="() => selectCategory('Company')">
          <i class="material-icons category-icon" style="padding:35px">work</i>
          <br />Company
        </div>
      </div>
    </div>
    <div class="row">
      <div class="column">
        <div class="category-column" @click="() => selectCategory('Me')">
          <i class="material-icons category-icon" style="padding:35px">face</i>
          <br />Me
        </div>
      </div>
      <div class="column">
        <div class="category-column" @click="() => selectCategory('Client')">
          <i class="material-icons category-icon" style="padding:35px"
            >domain</i
          >
          <br />Client
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import { mapActions } from "vuex";

export default {
  name: "FAQCategoryView",
  data() {
    return {
      category: ""
    };
  },

  methods: {
    selectCategory: function(category) {
      console.log(category);
      console.log(arguments);
      this.category = category;
      this.$store.dispatch("changeSearch", category);
      this.$store.dispatch("searchCategory");
      this.$router.push("/faq/search");
    }
  },

  computed: {
    ...mapActions(["changeSearch"])
  }
};
</script>

<style lang="scss">
.category-icon {
  font-size: 75px;
  opacity: 0.7;
}

.category {
  margin: 15px;
  font-family: "Poppins", sans-serif;
  font-style: normal;
  font-weight: bold;
  line-height: normal;
  text-align: center;
  color: #323232;
}

.row {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 100%;
}

.column {
  display: flex;
  padding: 1em;
  flex-direction: column;
  flex-basis: 100%;
  flex: 1;
}

.category-column {
  background-color: #fafafa;
  height: 11em;
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.24), 0px 0px 2px rgba(0, 0, 0, 0.12);
  border-radius: 2px;
}
</style>
