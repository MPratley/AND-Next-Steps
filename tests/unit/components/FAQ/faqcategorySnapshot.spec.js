import "jsdom-global/register";
import FAQCategoryView from "@/views/FAQ/FAQCategoryView.vue";
import Vue from "vue";
import { toMatchImageSnapshot } from "jest-image-snapshot";

expect.extend({ toMatchImageSnapshot });

const Constructor = Vue.extend(FAQCategoryView);
const vm = new Constructor().$mount();
describe("FAQCategoryView", () => {
  it("should render", () => {
    expect(vm.$el.querySelector("h2").textContent).toContain("Categories");
  });
  it("should match the snapshot", () => {
    expect(vm.$el).toMatchSnapshot();
  });

  it("renders correctly", async () => {
    jest.setTimeout(10000);
    const page = await browser.newPage();
    await page.goto("http://localhost:8080/#/faq/category");
    const image = await page.screenshot();

    expect(image).toMatchImageSnapshot();
  });
});
