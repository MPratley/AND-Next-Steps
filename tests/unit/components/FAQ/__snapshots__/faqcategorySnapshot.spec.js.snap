// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQCategoryView should match the snapshot 1`] = `
<div
  class="category"
>
  <h2
    class="title"
  >
    Categories
  </h2>
   
  <div
    class="row"
  >
    <div
      class="column"
    >
      <div
        class="category-column"
      >
        <i
          class="material-icons category-icon"
          style="padding: 35px;"
        >
          home
        </i>
         
        <br />
        Club
      
      </div>
    </div>
     
    <div
      class="column"
    >
      <div
        class="category-column"
      >
        <i
          class="material-icons category-icon"
          style="padding: 35px;"
        >
          work
        </i>
         
        <br />
        Company
      
      </div>
    </div>
  </div>
   
  <div
    class="row"
  >
    <div
      class="column"
    >
      <div
        class="category-column"
      >
        <i
          class="material-icons category-icon"
          style="padding: 35px;"
        >
          face
        </i>
         
        <br />
        Me
      
      </div>
    </div>
     
    <div
      class="column"
    >
      <div
        class="category-column"
      >
        <i
          class="material-icons category-icon"
          style="padding: 35px;"
        >
          domain
        </i>
         
        <br />
        Client
      
      </div>
    </div>
  </div>
</div>
`;
